client/login/controller.js

```diff
@@ -105,4 +106,18 @@ export default {
 
 		next();
 	},
+
+	redirectDefaultLocale( context, next ) {
+		// Do not redirect if user is logged in
+		if ( getCurrentUser( context.store.getState() ) ) {
+			return next();
+		}
+
+		// only redirect `/log-in/en` to `/log-in`
+		if ( context.pathname !== '/log-in/en' ) {
+			return next();
+		}
+
+		page.replace( '/log-in', context.query );
+	},
```

Comment:
Wouldn't it be possible to declare a route for `/log-in/en` in `index.node.js` instead of introducing this `redirectDefaultLocale` middleware? This would allow us to reuse what was done in `index.node.js`. Or maybe I'm missing something? 🙂

Reply:
The problem is that once we enter `/log-in/en` we can't just call `next()` to call the middlewares defined for `/login/:lang?`. So we need to handle all cases on the same route.